pkg/bindings/images/build.go

```diff
@@ -556,16 +557,26 @@ func nTar(excludes []string, sources ...string) (io.ReadCloser, error) {
 				merr = multierror.Append(merr, err)
 				return
 			}
-
 			err = filepath.Walk(s, func(path string, info os.FileInfo, err error) error {
 				if err != nil {
 					return err
 				}
 
-				if path == s {
-					return nil // skip root dir
+				// check if what we are given is an empty dir, if so then continue w/ it. Else return.
+				// if we are given a file or a symlink, we do not want to exclude it.
+				if info.IsDir() && s == path {
+					var p *os.File
+					p, err = os.Open(path)
+					if err != nil {
+						return err
+					}
+					defer p.Close()
+					_, err = p.Readdir(1)
+					if err != io.EOF || err != nil {
```

Comment:
sorry to butt in here but I'm following this PR and just noticed this so wanted to say something

seems like this conditional is redundant/not intended. This conditional always evaluates to true

1. err == nil (err != io.EOF)
2. err == io.EOF (err != nil)
3. err is anything else (err != nil)

Reply:
yeah you are right, nice catch. I will change it so that this statement is just used to catch !EOF and add another error check to log random errors